### ISSUE TITLE
[Site Design Revamp] Rotating a preview from portrait to landscape results in a blank screen

### ISSUE THREAD

**Issue by twstokes** [wordpress-mobile/WordPress-Android#16672]
### Expected behavior
Preview adapts to landscape orientation.

### Actual behavior
The preview shows a blank white screen.

![image](https://user-images.githubusercontent.com/2092798/171232209-6dd472a8-9931-41e2-8077-114a241264de.png)

### Steps to reproduce the behavior
1. Start the Site Creation flow
2. On the Design Picker screen in portrait orientation, tap a design to load the preview
3. Rotate to landscape
4. Observe that the preview is now blank (see screenshot)
5. Rotate back to portrait
6. Observe that the preview is still blank

**Comment by mkevins** [comment]
After investigating this a bit, I have found that the `selectedLayout` is null after an orientation change when we reach this line:

https://github.com/wordpress-mobile/WordPress-Android/blob/0abd1f3dfb5fb91574e76f2d69e1a191e6282150/WordPress/src/main/java/org/wordpress/android/ui/layoutpicker/LayoutPickerViewModel.kt#L253

Further upstream in the data flow, the saved state is not repopulated correctly here:

https://github.com/wordpress-mobile/WordPress-Android/blob/0abd1f3dfb5fb91574e76f2d69e1a191e6282150/WordPress/src/main/java/org/wordpress/android/ui/layoutpicker/LayoutPickerViewModel.kt#L308

I suspect because the `Bundle` being parsed there is not same as the one persisted here:

https://github.com/wordpress-mobile/WordPress-Android/blob/0abd1f3dfb5fb91574e76f2d69e1a191e6282150/WordPress/src/main/java/org/wordpress/android/ui/sitecreation/SiteCreationActivity.kt#L81

Tl;dr, the `hppViewModel` data is written to the `SiteCreationActivity` bundle, but after the orientation change, the view model data is loaded from the `HomePagePickerFragment`'s bundle:

https://github.com/wordpress-mobile/WordPress-Android/blob/0abd1f3dfb5fb91574e76f2d69e1a191e6282150/WordPress/src/main/java/org/wordpress/android/ui/sitecreation/theme/HomePagePickerFragment.kt#L60

**Comment by antonis** [comment]
fixed with https://github.com/wordpress-mobile/WordPress-Android/pull/16700
